packages/server/src/utilities/rowProcessor/index.ts

```diff
@@ -233,6 +232,11 @@ export async function outputProcessing<T extends Row[] | Row>(
       })
     : safeRows
 
+  // make sure squash is enabled if needed
+  if (!opts.squash && utils.hasCircularStructure(rows)) {
+    opts.squash = true
+  }
+
```

Comment:
If we can detect when this is needed, should it be an option at all?

Reply:
yes - squashing isn't just for removing circular dependencies, it can be used in a variety of different ways - if it was not possible to trigger it we would have to force circular dependencies to be able to squash relationships.